### ISSUE TITLE
Invalid field layout ID

### ISSUE THREAD

**Issue by mijewe** [verbb/gift-voucher#38]
### Description
I'm getting an "Invalid field layout ID" error when doing the following:
- viewing voucher codes (Admin > Gift Voucher > Voucher Codes). This reports a "A server error occured" and reports "Invalid field layout ID" in the logs.
- creating a new code (Admin > Gift Voucher > Voucher Codes > New Code)
- adding the voucher to my basket.

This was working fine on my local environment, but since deploying (and running `./craft project-config/sync`), there seems to be a mismatch of the layout IDs.

On my staging site, it reports "Invalid field layout ID: 51".
My staging "fieldlayouts" table does not have a record with ID 51.
However, my local "fieldlayouts" table does - of type "verbb\giftvoucher\elements\Code".

Locally,
The "fieldLayoutId" column for the voucher in my "giftvoucher_vouchertypes" has a value of 40 (not 51).
fieldlayouts 40 is of type "verbb\giftvoucher\elements\Voucher" - not sure how it all fits together but this sounds more like what it's looking for.

On staging,
The "fieldLayoutId" column for the voucher in my "giftvoucher_vouchertypes" has a value of 46 (not 51).
fieldlayouts 46 is also of type "verbb\giftvoucher\elements\Voucher"

To me it looks like it's trying to get the layoutid of a Code, not a Voucher Type, but that's as far as I've got.

Notable config:
```
'useProjectConfigFile' => true,
'allowUpdates' => (true in devmode, false otherwise),
'allowAdminChanges' => (true in devmode, false otherwise),
```

### Steps to reproduce

I'll try to put together a clean example when I have time, but my _assumed_ reproduction steps are

1. Create a new "Voucher Type" (x)
2. Add some fields to x's field layout.
3. Switch to a different database, and sync the project config
4. On your new environment, try to add a new voucher code at Admin > Gift Voucher > Voucher Codes > New Code.
5. Observe the "Invalid field layout ID" error.

### Additional info

- Plugin version: 2.3.0
- Craft version: Craft Pro 3.4.23
- Commerce version: 3.1.7

**Comment by mijewe** [comment]
(from our emails)
>  In the meantime, I’m wondering if you can enabled “allowAdminChanges” on your production or staging site, re-save the voucher type, and download the project.yaml file to your local dev.

I've done this but it didn't make a difference to the yaml.
I then added a new field to the layout to see if that would give it a kick, and imported my amended stage yaml into dev.
No luck - it's still trying to load Layout ID 51 (of type verbb\giftvoucher\elements\Code)

Thanks!

**Comment by engram-design** [comment]
So I've managed to figure this one out, just took a little longer as this code was added as part of a PR https://github.com/verbb/gift-voucher/pull/18.

This was related to the field layout for the code, managed in your plugin settings. It was incorrectly being saved, and not taking project config into account.

This should be all fixed in [2.3.1](https://github.com/verbb/gift-voucher/releases/tag/2.3.1)
